.github/workflows/typespec-latest-validation.yml

```diff
@@ -0,0 +1,110 @@
+name: TypeSpec Latest Validation
+
+on:
+  pull_request:
+    branches: [main, release/*]
+  issue_comment:
+    types: [created]
+  workflow_dispatch:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  typespec-latest-validation:
+    name: Validate against TypeSpec Latest
+    runs-on: ubuntu-latest
+    if: |
+      (github.event_name == 'workflow_dispatch') ||
+      (github.event_name == 'pull_request') ||
+      (github.event_name == 'issue_comment' && 
+       github.event.issue.pull_request &&
+       contains(github.event.comment.body, '/integration azure-specs') &&
+       (github.event.comment.author_association == 'MEMBER' || 
+        github.event.comment.author_association == 'OWNER' || 
+        github.event.comment.author_association == 'COLLABORATOR'))
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
+
+      - name: Checkout Azure REST API specs
+        uses: actions/checkout@v4
+        with:
+          repository: Azure/azure-rest-api-specs
+          path: azure-rest-api-specs
+          fetch-depth: 1
+
+      - name: Setup Node.js and pnpm
+        uses: ./.github/actions/setup
+
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Build and pack TypeSpec Azure packages
+        run: |
+          # Build the azure packages first
+          pnpm build || echo "Build completed with some issues, continuing with validation"
+          
+          # Create tgz packages for all TypeSpec Azure packages
+          echo "Creating tgz packages..."
+          
+          # Pack all TypeSpec Azure packages using chronus
+          pnpm chronus pack --pack-destination ./tgz-packages
+          
+          # List the created packages
+          echo "Created tgz packages:"
+          ls -la ./tgz-packages/
+        continue-on-error: true
+
+      - name: Update azure-rest-api-specs package.json
+        run: |
+          # Move to azure-rest-api-specs directory
+          cd azure-rest-api-specs
+          
+          # Update package.json with all dependencies and install
+          echo "Updating package.json with TypeSpec dependencies..."
+          pnpm tsx ../.github/scripts/update-azure-specs-packages.ts . ../tgz-packages
+          
+          echo "Installing dependencies..."
+          npm install
+        continue-on-error: true
+
+      - name: Run TypeSpec validation in azure-rest-api-specs
+        run: |
+          cd azure-rest-api-specs
```

Comment:
Convert this to a ts script instead. This also doesn't seem to discover all the entry points.

Logic probably should be find all folder with `tspconfig.yaml` and run tsp compile . in that folder